woocommerce/woocommerce
# GDPR: Order cleanup

For #19736 

Implements some settings to clean up failed/cancelled/pending orders at set intervals.

Adds the following settings, all off by default:

![slack - a8c 2018-04-18 12-07-49](https://user-images.githubusercontent.com/90977/38928585-331b36d4-4301-11e8-9a84-5be0c5377cd2.png)

Daily cron triggers a background process to run cleanup on matching orders.

Failed, cancelled, pending orders will be moved to the trash if they reach the criteria you set.

Competed guest orders will be anonymized if they reach the criteria you set.

To test:

1. After checkout out, deactivate and reactivate WC so the install script runs.
2. have a variety of orders some with:
- pending status - post date today and another from last month
- cancelled status - post date today and another from last month
- failed status - post date today and another from last month
- completed status - post date today and another from last month
3. Change all settings to target orders 1 week or older.
4. Run the cron job manually using WP Crontrol plugin. `woocommerce_cleanup_orders` event.

Check the orders. The older ones should have been trashed or anonymized based on types.
